### PR TITLE
Updated C28751

### PR DESCRIPTION
**Pull request by MugBergerFries**
Updated to include link to Microsoft announcement on the topic, added a list of banned functions, and matched the formatting to similar updates

### CODE REVIEW

File: windows-driver-docs-pr/devtest/28751-banned-exallocatepool-usage.md [hunk 1]
```diff
@@ -9,36 +9,16 @@ f1_keywords:
...
-</tr>
-</tbody>
-</table>
-
- 
-
+**warning C28751: Banned API Usage ExAllocatePool**\
+_Banned usage of ExAllocatePool and its variants: ExAllocatePool is a banned API for improved error handling purposes._ 
+
+This warning indicates that a function is being used that has been banned and has a more robust or secure replacement. This specific warning indicates the use of ExAllocatePool or one of its variants, which are deprecated. See [this article](Transition to ExAllocatePool2 API - OSGWiki ) for further information. A list of all banned functions covered by this error and recommended replacements can be found below:
```
Comment (MichaelSquires) [inline windows-driver-docs-pr/devtest/28751-banned-exallocatepool-usage.md:15]:
```suggestion
This warning indicates that a function is being used that has been banned and has a more robust or secure replacement. This specific warning indicates the use of ExAllocatePool or one of its variants, which are deprecated. See [this article](/windows-hardware/drivers/kernel/updating-deprecated-exallocatepool-calls) for further information. A list of all banned functions covered by this error and recommended replacements can be found below: 
```

For future reference, osgwiki is a Microsoft internal resource. It should not be used in public documentation.

File: windows-driver-docs-pr/devtest/28751-banned-exallocatepool-usage.md [hunk 2]
```diff
@@ -9,36 +9,16 @@ f1_keywords:
...
-
- 
-
+**warning C28751: Banned API Usage ExAllocatePool**\
+_Banned usage of ExAllocatePool and its variants: ExAllocatePool is a banned API for improved error handling purposes._ 
+
+This warning indicates that a function is being used that has been banned and has a more robust or secure replacement. This specific warning indicates the use of ExAllocatePool or one of its variants, which are deprecated. See [this article](Transition to ExAllocatePool2 API - OSGWiki ) for further information. A list of all banned functions covered by this error and recommended replacements can be found below: 
+
+## Banned Functions 
+| banned API | replacement(s) |
```
Comment (MichaelSquires) [inline windows-driver-docs-pr/devtest/28751-banned-exallocatepool-usage.md:18]:
```suggestion
| Banned API | Replacement Function(s) |
```

For the public documentation, please try to use proper formatting, grammar, punctuation, etc. I'd like these documents to be as polished and professional as possible.

File: windows-driver-docs-pr/devtest/28751-banned-exallocatepool-usage.md [hunk 3]
```diff
@@ -9,36 +9,16 @@ f1_keywords:
...
-<td align="left"><p><span id="ExAllocatePoolWithQuota"></span><span id="exallocatepoolwithquota"></span><span id="EXALLOCATEPOOLWITHQUOTA"></span>ExAllocatePoolWithQuota</p></td>
-<td align="left"><p>Secure replacement: <a href="/windows-hardware/drivers/ddi/wdm/nf-wdm-exallocatepoolwithquotatag" data-raw-source="[&lt;strong&gt;ExAllocatePoolWithQuotaTag&lt;/strong&gt;](/windows-hardware/drivers/ddi/wdm/nf-wdm-exallocatepoolwithquotatag)"><strong>ExAllocatePoolWithQuotaTag</strong></a> routine</p></td>
-</tr>
-</tbody>
-</table>
-
- 
-
+**warning C28751: Banned API Usage ExAllocatePool**\
+_Banned usage of ExAllocatePool and its variants: ExAllocatePool is a banned API for improved error handling purposes._
```
Comment (MichaelSquires) [inline windows-driver-docs-pr/devtest/28751-banned-exallocatepool-usage.md:13]:
```suggestion
_Banned usage of ExAllocatePool and its variants: The new APIs improve driver security by zeroing the returned allocation by default._ 
```
I think the major benefit of the new APIs are more about zeroing memory by default.

File: windows-driver-docs-pr/devtest/28751-banned-exallocatepool-usage.md [hunk 4]
```diff
@@ -9,36 +9,16 @@ f1_keywords:
...
-<tr class="even">
-<td align="left"><p><span id="ExAllocatePoolWithQuota"></span><span id="exallocatepoolwithquota"></span><span id="EXALLOCATEPOOLWITHQUOTA"></span>ExAllocatePoolWithQuota</p></td>
-<td align="left"><p>Secure replacement: <a href="/windows-hardware/drivers/ddi/wdm/nf-wdm-exallocatepoolwithquotatag" data-raw-source="[&lt;strong&gt;ExAllocatePoolWithQuotaTag&lt;/strong&gt;](/windows-hardware/drivers/ddi/wdm/nf-wdm-exallocatepoolwithquotatag)"><strong>ExAllocatePoolWithQuotaTag</strong></a> routine</p></td>
-</tr>
-</tbody>
-</table>
-
- 
-
+**warning C28751: Banned API Usage ExAllocatePool**\
```
Comment (MichaelSquires) [inline windows-driver-docs-pr/devtest/28751-banned-exallocatepool-usage.md:12]:
```suggestion
**Warning C28751: Banned API Usage ExAllocatePool**\
```

**Review by MichaelSquires — CHANGES_REQUESTED** [review]

File: windows-driver-docs-pr/devtest/28751-banned-exallocatepool-usage.md (hunk 3 above)
Comment (MugBergerFries) [reply windows-driver-docs-pr/devtest/28751-banned-exallocatepool-usage.md:13]: I used the text that shows up when the error actually occurs. If we want to change this, it should probably start by changing the actual error, then I can match the change here

Comment (MichaelSquires) [reply windows-driver-docs-pr/devtest/28751-banned-exallocatepool-usage.md:13]: Oh, I see. Sorry, that wasn't clear.

Can you add some text to make it clear that this line is the actual error message output from Prefast?

**Review by Rastaban — APPROVED** [review]

File: windows-driver-docs-pr/devtest/28751-banned-exallocatepool-usage.md (hunk 3 above)
Comment (MugBergerFries) [reply windows-driver-docs-pr/devtest/28751-banned-exallocatepool-usage.md]: Done!

**Review by MichaelSquires — APPROVED** [review]

**Review by Rastaban — APPROVED** [review]

**Comment by MugBergerFries** [comment]
@DOMARS This is approved and ready to be merged

**Pull request merged**